autograder/submitty_autograding_shipper.py

```diff
@@ -750,6 +751,17 @@ def checkout_vcs_repo(config, my_file, which_branch):
     checkout_path = os.path.join(course_dir, "checkout", partial_path)
     results_path = os.path.join(course_dir, "results", partial_path)
 
+    # Load the git branch for autgrading from the course config file
+    which_branch = 'undefined'
+    course_config_file = os.path.join(course_dir, "config", "config.json")
+    with open(course_config_file) as open_file:
+        COURSE_JSON = json.load(open_file)
+        if 'git_autograding_branch' in COURSE_JSON['course_details']:
+            which_branch = COURSE_JSON['course_details']['git_autograding_branch']
+    # verify that the branch only contains alphabetic characters a-z
+    if not re.match('^[a-z]+$', which_branch):
+        which_branch = 'main'
```

Comment:
I would at least allow numbers, capital letters, and underscores (and maybe hyphens?) into the regex.

Reply:
As I noted elsewhere, this field should really only be 'main' or 'master'.  
Or possibly another very simple string like 'default' if someone wants to go non-standard.
I intentionally picked lowercase only.  So I'm going to leave it for now rather than overdesign.
We can upgrade in a future PR if requested. Esp if/when we open it up as a config option on course settings UI.